File: Sample/go/src/tabletest/Int3.go
package tabletest
import (
    "scorpioproto"
)

// Int3 类名
type Int3 struct {
    
    _value1 *Int2
    _value2 int32
}


// ID   默认值()
func (data *Int3) ID() *Int2 { return data._value1; }

// Getvalue1   默认值()
func (data *Int3) Getvalue1() *Int2 { return data._value1; }

// Getvalue2   默认值()
func (data *Int3) Getvalue2() int32 { return data._value2; }


// GetData 获取数据
func (data *Int3) GetData(key string) interface{} {
    if "value1" == key {
        return data._value1;
    }
    if "value2" == key {
        return data._value2;
    }
    return nil;
}


// Set 设置数据
func (data *Int3) Set(value *Int3) {
    data._value1 = value._value1;
    data._value2 = value._value2;
}


// Read 读取数据
func (data *Int3) Read(fileName string, reader scorpioproto.IScorpioReader) {
    data._value1 = Int2Read(fileName, reader);
    data._value2 = reader.ReadInt32();
}

// Int3Read 读取数据
func Int3Read(fileName string, reader scorpioproto.IScorpioReader) *Int3 {
    ret := new(Int3)
    ret.Read(fileName, reader)
    return ret
}

File: Sample/ts/src/sco/Int3.ts
import { IScorpioReader } from '../ScorpioProto/Commons/IScorpioReader'
import { ScorpioUtil } from '../ScorpioProto/Commons/ScorpioUtil'
import { TableUtil } from '../ScorpioProto/Table/TableUtil'
import { IData } from '../ScorpioProto/Table/IData'
import { ITable } from '../ScorpioProto/Table/ITable'
import { Int2 } from './Int2'

export class Int3 implements IData {
    private m_IsInvalid:boolean = false;
    
    private _value1:any = null;
    /*   默认值() */
    getvalue1():Int2 { return this._value1; }
    ID():Int2 { return this._value1; }
    private _value2:any = null;
    /*   默认值() */
    getvalue2():number { return this._value2; }
    
    GetData(key:string):any {
        if ("value1" == key) return this._value1;
        if ("value2" == key) return this._value2;
        return null;
    }
    
    IsInvalid():boolean { return this.m_IsInvalid; }
    private CheckInvalid():boolean {
        return false;
    }
    
    Set(value:Int3) {
        this._value1 = value._value1;
        this._value2 = value._value2;
    }
    
    ToString() {
        return `{
            value1 : ${this._value1}, 
            value2 : ${this._value2}
            }`;
    }
    
    public static Read(fileName:string, reader:IScorpioReader):Int3 {
        var ret = new Int3();
        ret._value1 = Int2.Read(fileName, reader);
        ret._value2 = reader.ReadInt32();
        ret.m_IsInvalid = ret.CheckInvalid();
        return ret;
    }
}